src/components/ViewRequest/ViewRequest.jsx

```diff
@@ -0,0 +1,197 @@
+import { useEffect, useState } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useBackend } from '../../contexts/BackendContext';
+import { Box, Button, Card, ChakraProvider, Flex, Table, Heading, Thead, Tbody, Th, Tr, Td } from '@chakra-ui/react';
+
+const ViewRequest = () => {
+    const { backend } = useBackend();
+    const { id } = useParams();
+    const navigate = useNavigate();
+    const supplyItems = [
+        "Get Pet Food Decal",
+        "Decal",
+        "\"Homeless?\" Card",
+        "Business Card",
+        "Donation Site Decal",
+        "Donation Site Bin Decal",
+        "Donation Envelopes",
+        "\"Homesless?\" Card 2",
+    ]
+    const [itemAmounts, setItemAmounts] = useState({});
+    const [notes, setNotes] = useState("");
+    //const [status, setStatus] = useState(false);
+    const [businessName, setBusinessName] = useState("");
+    //const [notificationID, setNotificationID] = useState(0);
+    const [dateRequested, setDateRequested] = useState("");
+
+    const findBusinessId = (message) => {
+        const regex = new RegExp(`id: *[0-9]+`, 'i');
+        return message.match(regex, "i")[0].split(":")[1].trim();
+    }
+
+    function formatDate(dateTimeString) {
+        const date = new Date(dateTimeString);
+        const options = {
+          year: 'numeric',
+          month: '2-digit',
+          day: '2-digit',
+        };
+        return date.toLocaleDateString('en-US', options);
+      }
+
+    const fetchRequest = async () => {
+        try{
+            var temp ={};
+            const response = await backend.get(`/notification/request/${id}`);
+            const message = response.data[0].message;
+            const lines = message.split("\n");
+            supplyItems.map((item, index) => {
+                const matchAmount = lines[index + 1].split(": ")[1];
+                temp[item] = matchAmount;
+            });
+            //setNotificationID(response.data[0].notification_id);
+            //setStatus(response.data[0].been_dismissed);
+            setDateRequested(formatDate(response.data[0].timestamp));
+            setNotes(lines.slice(11).join("\n"));
+            setItemAmounts({...temp});
+            const businessId = findBusinessId(message);
+            const businessResponse = await backend.get(`/business/${businessId}`);
+            setBusinessName(businessResponse.data[0].name);
+        }
+        catch(error){
+            console.error(error);
+        }
+        
+    }
+    useEffect(() => {
+        fetchRequest();
+    }, [id]);

```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'fetchRequest'. Either include it or remove the dependency array.